### PR TITLE
Update drone.yml workspace path and test step

### PR DESCRIPTION
**Pull request by fernandrone**
Hardcode path, so `drone exec` will work in any environment.

Make go test iterate through every package, so it runs coverage in all of them, instead of just testing the root folder (which incidentally contains no test files).

### CODE REVIEW

**Comment by fernandrone** [comment]
Before:

image

Now:

image

**Comment by fernandrone** [comment]
Also, it seems drone did not run after the last push. Maybe it's been configured to run on PR, but not on push (which I guess would be the preferred behavior)?

File: .drone.yml [hunk 1]
```diff
@@ -8,7 +9,12 @@ pipeline:
       - CGO_ENABLED=0
     commands:
       - go vet
-      - go test -cover -coverprofile=coverage.out
+      - |
+        for line in $(go list ./... | grep -v /vendor/)
+        do  
+          name=$(basename "$line")
+          go test "$line" -cover -coverprofile=coverage-"$name".out
```
Comment (tboerger) [inline .drone.yml]: Do it like https://github.com/drone-plugins/drone-coverage/blob/release/0.4/Makefile#L25

File: .gitignore [hunk 2]
```diff
@@ -24,5 +24,7 @@ _testmain.go
 *.prof
 .env
 
-coverage.out
+*.swp
```
Comment (tboerger) [inline .gitignore]: This should not be part of the gitignore, that should be handled on your own global gitignore.

File: .gitignore [hunk 3]
```diff
@@ -24,5 +24,7 @@ _testmain.go
...
+coverage*.out
```
Comment (tboerger) [inline .gitignore]: Just drop that change, the old ignore will be matched if you do it like the old makefile.

File: .gitignore [hunk 4]
```diff
@@ -24,5 +24,7 @@ _testmain.go
...
+coverage.xml
```
Comment (tboerger) [inline .gitignore]: Where does the xml format come from?

**Review by tboerger — CHANGES_REQUESTED** [review]

File: .drone.yml (hunk 1 above)
Comment (fernandrone) [reply .drone.yml]: Ok, will do!

File: .gitignore (hunk 2 above)
Comment (fernandrone) [reply .gitignore]: Ok

File: .gitignore (hunk 3 above)
Comment (fernandrone) [reply .gitignore]: Alright

File: .gitignore (hunk 4 above)
Comment (fernandrone) [reply .gitignore]: Yeah, I can remove this. Cobertura uses xml files so I added a few on my local environment for testing, but it doesn't need to go on .gitignore, my bad.

**Comment by fernandrone** [comment]
@tboerger updated PR, please check if it's ok now :+1:

**Review by tboerger — APPROVED** [review]

**Comment by tboerger** [comment]
LGTM

File: .drone.yml [hunk 5]
```diff
@@ -8,6 +9,11 @@ pipeline:
       - CGO_ENABLED=0
     commands:
       - go vet
+      - |
+        for pkg in $(go list ./... | grep -v /vendor/)
+        do  
+          go test -cover -coverprofile $GOPATH/src/$pkg/coverage.out $pkg || exit 1
+        done
       - go test -coverprofile=coverage.out
```
Comment (donny-dont) [inline .drone.yml]: This line can be gotten rid of correct?

**Review by donny-dont — APPROVED** [review]

**Comment by donny-dont** [comment]
LGTM

**Pull request merged**